lib/sunrise/sunrise.php

```diff
@@ -66,6 +66,13 @@ function handle_not_found_error( $error_type ) {
 		exit;
 	}
 
+	// Empty get_sites() return would signify potential cache pollution
+	// We side-step by bumping the last changed which would result in the cache key change
+	// So the next request can recover
+	if ( 'site' == $error_type && ! get_sites() ) {
+		wp_cache_set_sites_last_changed();
```

Comment:
`get_sites()` could be an unbounded query on a large MS. Since really we just want to make sure we aren't involved with cache pollution, maybe lets just check the cache itself: https://github.com/WordPress/WordPress/blob/97ba08426dd1ca896596646e708b2d17387ed287/wp-includes/class-wp-site-query.php#L357-L361

Could check the key that is used when `$args` is empty to get the same result you're currently doing, just without the extra queries running at the end.